### ISSUE TITLE
[request/suggestion] new release

### ISSUE THREAD

**Issue by macearl** [BenoitDuffez/AndroidCupsPrint#109]
Hi, as there are tons of issues recently closed it would be great if you could build a new release apk, or mark a branch as a new release (whatever makes fdroid build a new version)

I've installed your app on quite a few devices (most of them are not mine) and it would make it a lot easier to update them if i could just tell people to refresh their f-droid repositories instead of making them uninstall the app and reinstall a self build one instead.

cheers

**Comment by BenoitDuffez** [comment]
I have some commits pending, I'll push them in a few minutes. I'll release a beta with that. Once it's stable I'll push it to prod.

However f-droid is taking me a lot of time and sweat, and I use some proprietary dependencies (on top of my head Crashlytics, don't know if there are more) so I guess that's not a simple push to get a release on f-droid.

If you are knowledgeable about f-droid, feel free to fork/PR for f-droid and I'll merge.

**Comment by BenoitDuffez** [comment]
OK now everything should be automatic. I have a cron job that runs at 2am every day which makes a new RC release on beta branch, based off of the last commit on develop.

So any commit sent to develop will be released as a RC in the beta branch.  
And at the same time the beta branch is compiled and uploaded to the Play Store.

So basically any commit pushed to develop will be published as beta on the Play Store the next day.

And last, any merge to master will result in a non-RC release and a production build.

So for today I have bumped the version to 1.5.0 (from 1.4.0) and the build that will be made today will be 1.5.0-RC1. If another commit comes by later on, it will create 1.5.0-RC2, and so on.  
The merge on master will generate the 1.5.0 final.

I hope that's what you wanted. Sorry about f-droid but I don't have time to merge on that branch.

**Comment by BenoitDuffez** [comment]
OK that went well! Subscribe to the beta version of the app on the Play Store to receive all what I just pushed. :)

![screen shot 2018-09-07 at 02 11 40](https://user-images.githubusercontent.com/802209/45191435-7fb63280-b243-11e8-97b2-e534594c929f.png)

**Comment by macearl** [comment]
Thanks, i will give it a try, sadly i am not familiar with F-Droids build process

**Comment by IzzySoft** [comment]
If @BenoitDuffez could update 8the fdroid branch}(https://github.com/BenoitDuffez/AndroidCupsPrint/tree/fdroid), we (at F-Droid) might be able to build an update. Master includes some proprietary libraries which are not permitted in #F-Droid (like Crashlytics). Alternatively, if Crashlytics could be replaced by ACRA or Matomo, a separate branch might not even be required …

**Comment by BenoitDuffez** [comment]
Pull requests welcome.

**Comment by IzzySoft** [comment]
*Sigh* well… unfortunately, not being a developer, I cannot help with that… But good to know you're open to it!

**Comment by BenoitDuffez** [comment]
No worries, not everyone is. Now that you mentioned this issue, it's a little bit higher in the list (at least in my head!). Unfortunately I can't allow as much time as I want on this project.

**Comment by IzzySoft** [comment]
Yes, I know that problem – one can fork a project, but not the developer (they can "spawn", but it takes several years for productive results 🤣 )…

So I take it the Debian way: it's available when it's ready. Thanks!

**Comment by leela52452** [comment]
can you please relook into the issue
thank you

**Comment by BenoitDuffez** [comment]
I don't have time. I will accept pull requests.

**Comment by jugendhacker** [comment]
I tried to solve the problem with my PR but please have a look at it if it fits right.....

**Comment by BenoitDuffez** [comment]
OK I have attempted to update my gitlab repo and merge it to gitlab.com:fdroiddata/master with the newest 1.5.0 release, pointing to the latest commit of the f-droid branch of this repo.

Here it is: https://gitlab.com/fdroid/fdroiddata/merge_requests/5967/pipelines

My branch on that repo (on gitlab) was 18k commits behind, I attempted to rebase my work and update the yml. The pipeline passed, but I have zero idea whether what I did was good or not.

Anyway if someone has a good knowledge of how fdroid works, please hint me or submit PR here or on the gitlab repo.

**Comment by doronbehar** [comment]
@BenoitDuffez thanks for the effort. Perhaps you shouldn't have marked it as WIP ? I think that's why it received no comments/attention from the F Droid guys..

**Comment by licaon-kter** [comment]
So you've made a fdroid branch, great...but you add non-FOSS Crashlytics to it? Why?

**Comment by BenoitDuffez** [comment]
I didn't _add_ non FOSS to the branch: I didn't _remove_ it from the master branch. This app on production on the play store has crashlytics.

Why? I think I made it pretty clear in all the recent issues that I don't have time to invest in this project.

It's FOSS, you are free to clone it, remove what you don't like and install whatever you want on your device.

**Comment by licaon-kter** [comment]
@BenoitDuffez FYI https://gitlab.com/fdroid/fdroiddata/-/merge_requests/6420
